### PR TITLE
Use locked_requirement platform to print it out

### PR DESCRIPTION
**Pull request by indirect**

### CODE REVIEW

**Comment by indirect** [comment]
@segiddins r?

File: lib/bundler/resolver.rb [hunk 1]
```diff
@@ -3,12 +3,20 @@ class Resolver
     require "bundler/vendored_molinillo"
 
     class Molinillo::VersionConflict
+      def printable_dep(dep)
+        case dep
+        when DepProxy; dep.to_s.strip
+        when Bundler::Dependency; DepProxy.new(dep, dep.platforms.join(", ")).to_s.strip
+        else; raise("Don't know how to print #{dep.inspect}")
```
Comment (segiddins) [inline lib/bundler/resolver.rb]: Gem::Dep?

Comment (indirect) [inline lib/bundler/resolver.rb]: It seems we only ever have a Bundler::Dep (which is what defines #platforms) in our suite.

Comment (segiddins) [inline lib/bundler/resolver.rb]: I don't see a need to raise here, could fallback to `dep.to_s.strip` for everything but a `Bundler::Dependency` ?

Comment (indirect) [inline lib/bundler/resolver.rb]: Sure, works for me.

**Comment by segiddins** [comment]
@homu r+

**Comment by indirect** [comment]
@homu r=segiddins

**Comment by segiddins** [comment]
@homu r+

**Comment by indirect** [comment]
@homu retry

**Pull request merged**